libpod/container_internal_linux.go

```diff
@@ -773,6 +773,18 @@ func (c *Container) generateSpec(ctx context.Context) (*spec.Spec, error) {
 		}
 	}
 
+	// Pass down the LISTEN_* environment (see #10443).
+	for _, key := range []string{"LISTEN_PID", "LISTEN_FDS", "LISTEN_FDNAMES"} {
+		if val, ok := os.LookupEnv(key); ok {
+			// Force the PID to `1` since we cannot rely on (all
+			// versions of) all runtimes to do it for us.
+			if key == "LISTEN_PID" {
+				val = "1"
```

Comment:
What if the user passed --init, then the main pid will be 2. Who is supposed to rewrite that. For sure catatonit (currently used on e.g. fedora) doesn't do that.

Reply:
@alexlarsson This is been fixed in underlying runtime via https://github.com/containers/crun/pull/721 if **catatonit** sets `LISTEN_PID=2` crun will adhere it. We can remove this explicit `LISTEN_PID=1` once latest crun is released and vendor-ed into the Podman CI.

However following edge case still remains if OCI runtime is `runc` this is only fixed for `crun`. cc @giuseppe

Reply:
Fair point, @ alexlarsson.  I think Podman should set it since we should not (and to a certain degree cannot) rely on the runtimes or inits to do this work.

Mind opening a PR against Podman?

Reply:
ftr, @giuseppe is fixing catatonit here: https://github.com/openSUSE/catatonit/pull/15